packages/query-sequelize/src/query/sql-comparison.builder.ts

```diff
@@ -48,6 +48,40 @@ export class SQLComparisionBuilder<Entity> {
       // comparison operator (e.b. =, !=, >, <)
       return { [col]: { [this.comparisonMap[normalizedCmp]]: val } };
     }
+    if (normalizedCmp === 'between') {
+      // between comparison (field BETWEEN x AND y)
+      return this.betweenComparisonSQL(col, val);
+    }
+    if (normalizedCmp === 'notbetween') {
+      // notBetween comparison (field NOT BETWEEN x AND y)
+      return this.notBetweenComparisonSQL(col, val);
+    }
     throw new Error(`unknown operator "${cmp}"`);
   }
+
+  private betweenComparisonSQL<F extends keyof Entity>(
+    col: string,
+    val: EntityComparisonField<Entity, F>,
+  ): WhereOptions {
+    // TODO: this is not very good - can we better type this?!
+    // eslint-disable-next-line
+    const value = val as any;
```

Comment:
Yeah this isnt the greatest :( We'll need to play with the type def above. For now I think the following should work.

```suggestion
const value = (val as unknown) as CommonFieldComparisonBetweenType<Entity[F]>;
```